systemd/systemd.go

```diff
@@ -1064,9 +1112,17 @@ LazyUnmount=yes
 WantedBy=multi-user.target
 `
 
-func writeMountUnitFile(snapName, revision, what, where, fstype string, options []string) (mountUnitName string, err error) {
-	content := fmt.Sprintf(mountUnitTemplate, snapName, revision, what, where, fstype, strings.Join(options, ","))
-	mu := MountUnitPath(where)
+func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+	snapDesc := u.SnapName
+	if u.Revision != "" {
+		snapDesc += fmt.Sprintf(", revision %s", u.Revision)
+	}
+	content := fmt.Sprintf(mountUnitTemplate, snapDesc,
+		u.What, u.Where, u.Fstype, strings.Join(u.Options, ","))
+	if u.Creator != "" {
+		content += fmt.Sprintf("X-SnappyCreatorModule=%s\n", u.Creator)
```

Comment:
ah I see you already added this var, nvm my earlier comment about doing it this way